nav2_planner/src/planner_server.cpp

```diff
@@ -386,17 +376,14 @@ PlannerServer::computePlanThroughPoses()
       curr_goal = goal->goals[i];
 
       // Transform them into the global frame
-      if (!transformPosesToGlobalFrame(action_server_poses_, curr_start, curr_goal)) {
-        return;
+      if (!transformPosesToGlobalFrame(curr_start, curr_goal)) {
+        throw nav2_core::PlannerTFError("Unable to transform poses to global frame");
       }
 
       // Get plan from start -> goal
       nav_msgs::msg::Path curr_path = getPlan(curr_start, curr_goal, goal->planner_id);
 
-      // check path for validity
-      if (!validatePath(action_server_poses_, curr_goal, curr_path, goal->planner_id)) {
-        return;
-      }
+      validatePath<ActionThroughPoses>(curr_goal, curr_path, goal->planner_id);
```

Comment:
We have a few functions here of the style `if (!function(args)) {throw ...}`, we should do that here too. Have validate return a bool that throws here, not in the function body, to be consistent